### DISCUSSION TITLE
Track requirements.txt

### DISCUSSION THREAD

**Discussion by gshotwell** [posit-dev/positron#4376] [Feedback]
A common problem that I see a lot (especially with deploying to Posit Connect), is that user `requirements.txt` files are too extensive. They're often generated with `pip freeze` which locks down a lot of unnecessary dependencies and sometimes makes them impossible to resolve on another operating system. It would be nice if positron had a package installer which also managed a `requirements.txt` file. How this would work is that the installer would only write the top level packages and their versions to `requirements.txt` when they were installed. This would keep the file clean and give Connect a better chance at resolving the environment.

**Comment by juliasilge** [comment]
Thanks for sharing this, especially regarding that common pain point! We have some ideas for "next level" environment management in #1391 and I'll add a link there to this idea.
